Makefile

```
@@ -27,36 +51,48 @@ endif
+format:
+	./format.sh
+
+# helper commands
+update-box:
+	@SSL_CERT_FILE=${SSL_CERT_FILE} CURL_CA_BUNDLE=${CURL_CA_BUNDLE} vagrant box update || (echo '\n\nIf you get an SSL error you might be behind a transparent proxy. \nMore info https://github.com/fredrikhgrelland/vagrant-hashistack/blob/master/README.md#if-you-are-behind-a-transparent-proxy\n\n' && exit 2)
+
+proxy-to-minio:
+	consul connect proxy -service minio-local -upstream minio:9000 -log-level debug
+docker-local-build:
+	docker build docker -t fredrikhgrelland/hive:${LAST_COMMIT_HASH}
+	docker tag fredrikhgrelland/hive:${LAST_COMMIT_HASH} fredrikhgrelland/hive:test
+
+template_example: custom_ca
 ifdef CI # CI is set in Github Actions
-	cd test_example; SSL_CERT_FILE=${SSL_CERT_FILE} CURL_CA_BUNDLE=${CURL_CA_BUNDLE} vagrant up --provision
+	cd template_example; SSL_CERT_FILE=${SSL_CERT_FILE} CURL_CA_BUNDLE=${CURL_CA_BUNDLE} vagrant up --provision
 else
-	if [ -f "docker/conf/certificates/*.crt" ]; then cp -f docker/conf/certificates/*.crt test_example/docker/conf/certificates; fi
-	cd test_example; SSL_CERT_FILE=${SSL_CERT_FILE} CURL_CA_BUNDLE=${CURL_CA_BUNDLE} CUSTOM_CA=${CUSTOM_CA} ANSIBLE_ARGS='--extra-vars "local_test=true"' vagrant up --provision
+	if [ -f "docker/conf/certificates/*.crt" ]; then cp -f docker/conf/certificates/*.crt template_example/docker/conf/certificates; fi
+	cd template_example; SSL_CERT_FILE=${SSL_CERT_FILE} CURL_CA_BUNDLE=${CURL_CA_BUNDLE} CUSTOM_CA=${CUSTOM_CA} ANSIBLE_ARGS='--extra-vars "local_test=true"' vagrant up --provision
 endif
 
-# clean commands
-destroy:
-	vagrant destroy -f
-	rm terraform.tfstate || true
-	rm terraform.tfstate.backup || true
-	rm example/terraform.tfstate || true
 status:
 	vagrant global-status
-format:
-	./format.sh
+
+# clean commands
+destroy-box:
+	vagrant destroy -f
 
 remove-tmp:
 	rm -rf ./tmp
+	rm -rf ./.vagrant
+	rm -rf ./.minio.sys
+	rm -rf ./example/.terraform
+	rm -rf ./example/terraform.tfstate
+	rm -rf ./example/terraform.tfstate.backup
 
-clean: destroy remove-tmp
+clean: destroy-box remove-tmp
 
 # helper commands
 update-box:
-	@SSL_CERT_FILE=${SSL_CERT_FILE} CURL_CA_BUNDLE=${CURL_CA_BUNDLE} vagrant box update || (echo '\n\nIf you get an SSL error you might be behind a transparent proxy. \nMore info https://github.com/fredrikhgrelland/vagrant-hashistack/blob/master/README.md#if-you-are-behind-a-transparent-proxy\n\n' && exit 2)
+	@SSL_CERT_FILE=${SSL_CERT_FILE} CURL_CA_BUNDLE=${CURL_CA_BUNDLE} vagrant box update || (echo '\n\nIf you get an SSL error you might be behind a transparent proxy. \nMore info https://github.com/fredrikhgrelland/vagrant-hashistack/blob/master/README.md#proxy\n\n' && exit 2)
 
-proxy-to-minio:
-	consul connect proxy -service minio-local -upstream minio:9000 -log-level debug
-docker-local-build:
-	docker build docker -t fredrikhgrelland/hive:${LAST_COMMIT_HASH}
-	docker tag fredrikhgrelland/hive:${LAST_COMMIT_HASH} fredrikhgrelland/hive:test
+pre-commit: check_for_docker_binary
```

```suggestion
pre-commit: check_for_docker_binary check_for_terraform_binary
```
Should also be fixed in template